cmd/dep/ensure.go

```diff
@@ -145,17 +210,11 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 			Constraint: pc.Constraint,
 		}
 
-		if p.Lock != nil {
-			for i, lp := range p.Lock.P {
-				if lp.Ident() == pc.Ident {
-					p.Lock.P = append(p.Lock.P[:i], p.Lock.P[i+1:]...)
-					break
-				}
-			}
-		}
+		// Ignore the lockfile for this dependency and allow its version to change
+		params.ToChange = append(params.ToChange, pc.Ident.ProjectRoot)
```

Comment:
I think it's probably better not to tip the flag in this case. If the constraint has changed in a way that prohibits the version in the lock, it'll work itself out, anyway. Otherwise, the user didn't say `-update`, so...